### PR TITLE
fix(anta.tests): Cleaning up STUN tests module

### PR DESCRIPTION
**Pull request by vitthalmagadum**
# Description

Refactoring STUN tests module to address the following issues:

- https://github.com/aristanetworks/anta/issues/809
- https://github.com/aristanetworks/anta/issues/587
- https://github.com/aristanetworks/anta/issues/388

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have run pre-commit for code linting and typing (`pre-commit run`)
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/stun.py [hunk 1]
```diff
@@ -74,42 +70,24 @@ def test(self) -> None:
...
             bindings = command.json_output["bindings"]
-            source_address = str(command.params.source_address)
-            source_port = command.params.source_port
+            input_public_address = client_input.public_address
+            input_public_port = client_input.public_port
 
             # If no bindings are found for the STUN client, mark the test as a failure and continue with the next client
             if not bindings:
-                self.result.is_failure(f"STUN client transaction for source `{source_address}:{source_port}` is not found.")
+                self.result.is_failure(f"{client_input} - STUN client transaction not found.")
```
Comment (gmuloc) [inline anta/tests/stun.py:78]: This was an error from the first test
```suggestion
                self.result.is_failure(f"{client_input} - STUN client translation not found.")
```

```
site3-wan1#show stun client ?
  counters      Display STUN client counters
  translations  Show STUN client translations
```

File: anta/tests/stun.py [hunk 2]
```diff
@@ -7,23 +7,29 @@
...
+from anta.tools import get_value
 
 
 class VerifyStunClient(AntaTest):
-    """Verifies STUN client settings, including local IP/port and optionally public IP/port.
+    """Verifies the STUN client configuration.
+
+    This test performs the following checks for each specified address family:
+
+      1. Validates that the STUN client is configured.
```
Comment (gmuloc) [inline anta/tests/stun.py:22]:
```suggestion
      1. Validates that there is a translation for the STUN client.
```

File: anta/tests/stun.py [hunk 3]
```diff
@@ -74,42 +70,24 @@ def test(self) -> None:
...
             transaction_id = next(iter(bindings.keys()))
 
-            # Prepare the actual and expected STUN data for comparison
-            actual_stun_data = {
-                "source ip": get_value(bindings, f"{transaction_id}.sourceAddress.ip"),
-                "source port": get_value(bindings, f"{transaction_id}.sourceAddress.port"),
-            }
-            expected_stun_data = {"source ip": source_address, "source port": source_port}
-
             # If public address is provided, add it to the actual and expected STUN data
```
Comment (gmuloc) [inline anta/tests/stun.py:84]: comment is not correct anymore

File: anta/tests/stun.py [hunk 4]
```diff
@@ -74,42 +70,24 @@ def test(self) -> None:
...
-            if public_address is not None:
-                actual_stun_data["public ip"] = get_value(bindings, f"{transaction_id}.publicAddress.ip")
-                expected_stun_data["public ip"] = str(public_address)
+            if input_public_address and str(input_public_address) != (actual_public_address := get_value(bindings, f"{transaction_id}.publicAddress.ip")):
+                self.result.is_failure(f"{client_input} - Incorrect public-facing address; Expected: {input_public_address} Actual: {actual_public_address}")
 
             # If public port is provided, add it to the actual and expected STUN data
```
Comment (gmuloc) [inline anta/tests/stun.py:88]: same regarding tejh comment

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: anta/tests/stun.py (hunk 1 above)
Comment (vitthalmagadum) [reply anta/tests/stun.py]: Updated

File: anta/tests/stun.py (hunk 2 above)
Comment (vitthalmagadum) [reply anta/tests/stun.py]: updated

File: anta/tests/stun.py (hunk 3 above)
Comment (vitthalmagadum) [reply anta/tests/stun.py]: updated

File: anta/tests/stun.py (hunk 4 above)
Comment (vitthalmagadum) [reply anta/tests/stun.py]: updated

File: anta/tests/stun.py [hunk 5]
```diff
@@ -7,23 +7,29 @@
...
     Expected Results
     ----------------
-    * Success: The test will pass if the STUN client is correctly configured with the specified IPv4 source address/port and public address/port.
-    * Failure: The test will fail if the STUN client is not configured or if the IPv4 source address, public address, or port details are incorrect.
+    * Success: If all of the following conditions are met:
+        - The test will pass if the STUN client is correctly configured.
+        - If public IP and port details are provided, they must also match the configuration.
+    * Failure: If any of the following occur:
+        - The STUN client is not configured.
+        - The public IP or port details, if specified, are incorrect.
```
Comment (gmuloc) [inline anta/tests/stun.py:32]:
```suggestion
        - The test will pass if the STUN client translation is present.
        - If public IP and port details are provided, they must also match the translation information.
    * Failure: If any of the following occur:
        - There is not translation for the STUN client.
        - The public IP or port details, if specified, are incorrect.
```

File: anta/tests/stun.py [hunk 6]
```diff
@@ -7,23 +7,29 @@
...
-from anta.custom_types import Port
+from anta.input_models.stun import ClientAddress
 from anta.models import AntaCommand, AntaTemplate, AntaTest
-from anta.tools import get_failed_logs, get_value
+from anta.tools import get_value
 
 
 class VerifyStunClient(AntaTest):
-    """Verifies STUN client settings, including local IP/port and optionally public IP/port.
+    """Verifies the STUN client configuration.
```
Comment (gmuloc) [inline anta/tests/stun.py:18]:
```suggestion
    """Verifies the translation for a source address on a STUN client.
```

File: anta/tests/stun.py [hunk 7]
```diff
@@ -7,23 +7,29 @@
...
+
+    This test performs the following checks for each specified address family:
+
+      1. Validates that there is a translation for the STUN client.
```
Comment (gmuloc) [inline anta/tests/stun.py:22]:
```suggestion
      1. Validates that there is a translation for the source address on the STUN client.
```

File: anta/input_models/stun.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+from __future__ import annotations
+
+from ipaddress import IPv4Address
+
+from pydantic import BaseModel, ConfigDict
+
+from anta.custom_types import Port
+
+
+class ClientAddress(BaseModel):
```
Comment (gmuloc) [inline anta/input_models/stun.py:15]: Going though the test I think we should call this StunClientTranslation

Comment (vitthalmagadum) [reply anta/input_models/stun.py]: Updated test name as well.
Thanks!

File: anta/tests/stun.py [hunk 9]
```diff
@@ -7,29 +7,35 @@
...
-from pydantic import BaseModel
-
-from anta.custom_types import Port
+from anta.input_models.stun import StunClientTranslation
 from anta.models import AntaCommand, AntaTemplate, AntaTest
-from anta.tools import get_failed_logs, get_value
+from anta.tools import get_value
+
 
+class VerifyStunClientTranslation(AntaTest):
```
Comment (gmuloc) [inline anta/tests/stun.py:17]: to be backward compatible (SemVer) lets add VerifyStunClient with the deprecation warning (you have a decorator for this)

File: anta/tests/stun.py [hunk 10]
```diff
@@ -43,73 +50,72 @@ class VerifyStunClient(AntaTest):
...
+                self.result.is_failure(f"{client_input} - Incorrect public-facing address - Expected: {input_public_address} Actual: {actual_public_address}")
+
+            # Verifying the public port if provided
+            if input_public_port and input_public_port != (actual_public_port := get_value(bindings, f"{transaction_id}.publicAddress.port")):
+                self.result.is_failure(f"{client_input} - Incorrect public-facing port - Expected: {input_public_port} Actual: {actual_public_port}")
+
+
+@deprecated_test_class(new_tests=["VerifyStunClientTranslation"])
+class VerifyStunClient(VerifyStunClientTranslation):
+    """(deprecated) Verifies the translation for a source address on a STUN client.
```
Comment (gmuloc) [inline anta/tests/stun.py:96]:
```suggestion
    """(Deprecated) Verifies the translation for a source address on a STUN client.
```

**Review by gmuloc — APPROVED** [review]
LGTM once CI and pre-commit are happy

**Pull request merged**